app/jobs/inboxes/fetch_imap_email_inboxes_job.rb

```diff
@@ -2,8 +2,19 @@ class Inboxes::FetchImapEmailInboxesJob < ApplicationJob
   queue_as :scheduled_jobs
 
   def perform
+    # check imap_enabled for channel
     Inbox.where(channel_type: 'Channel::Email').all.find_each(batch_size: 100) do |inbox|
-      ::Inboxes::FetchImapEmailsJob.perform_later(inbox.channel) if inbox.channel.imap_enabled
+      next unless inbox.channel.imap_enabled?
+
+      fetch_emails(inbox)
+    end
+  end
+
+  def fetch_emails(inbox)
+    if inbox.channel.microsoft? && ENV.fetch('AZURE_TENANT_ID', false)
+      ::Inboxes::FetchMsGraphEmailsJob.perform_later(inbox.channel)
```

Comment:
`fetchMsGraphEmailForTenantJob` ?  in the name to depicts this works on for the case of single tenant ?